sdk/iot/hub/samples/methods/src/iot_hub_methods_example.c

```diff
@@ -0,0 +1,430 @@
+  }
+  *out = in * 2;
+  return AZ_OK;
+}
+
+static az_result send_method_response(
+    az_iot_hub_client_method_request* request,
+    uint16_t status,
+    int64_t return_value)
+{
+  az_result result;
+
+  // Get the response topic
+  char methods_response_topic[128];
+  if ((result = az_iot_hub_client_methods_response_get_publish_topic(
+           &client,
+           request->request_id,
+           status,
+           methods_response_topic,
+           sizeof(methods_response_topic),
+           NULL))
+      != AZ_OK)
+  {
+    printf("Unable to get twin document publish topic, return code %d\n", result);
+    return result;
+  }
+
+  // Build the response payload
+  az_span response_span;
+  if (status == 200)
+  {
+    az_json_builder json_builder;
+    result = az_json_builder_init(&json_builder, AZ_SPAN_FROM_BUFFER(method_reponse_payload));
+    result = az_json_builder_append_token(&json_builder, az_json_token_object_start());
+    az_json_token reported_property_value_token = az_json_token_number((double)return_value);
+    result = az_json_builder_append_object(
+        &json_builder, method_response_name, reported_property_value_token);
+    result = az_json_builder_append_token(&json_builder, az_json_token_object_end());
+
+    response_span = az_json_builder_span_get(&json_builder);
+  }
+  else
+  {
+    response_span = AZ_SPAN_FROM_STR("{}");
+  }
+
+  printf("Status: %u\tPayload:", status);
+  char* payload_char = (char*)az_span_ptr(response_span);
+  for (int32_t i = 0; i < az_span_size(response_span); i++)
+  {
+    putchar(*(payload_char + i));
+  }
+  putchar('\n');
+
+  // Send the response
+  if ((result = MQTTClient_publish(
+           mqtt_client,
+           methods_response_topic,
+           az_span_size(response_span),
+           az_span_ptr(response_span),
+           0,
+           0,
+           NULL))
+      != MQTTCLIENT_SUCCESS)
+  {
+    printf("Failed to publish twin document request, return code %d\n", result);
+    return result;
+  }
+
+  printf("Sent response\n");
+
+  return result;
+}
+
+static int on_received(void* context, char* topicName, int topicLen, MQTTClient_message* message)
+{
+  (void)context;
+
+  if (topicLen == 0)
+  {
+    // The length of the topic if there are one or more NULL characters embedded in topicName,
+    // otherwise topicLen is 0.
+    topicLen = (int)strlen(topicName);
+  }
+
+  az_iot_hub_client_method_request method_request;
+  if (az_iot_hub_client_methods_parse_received_topic(
+          &client, az_span_init((uint8_t*)topicName, topicLen), &method_request)
+      == AZ_OK)
+  {
+    az_result result;
+
+    printf("Direct Method arrived\n");
+    if (az_span_is_content_equal(double_method_span, method_request.name))
+    {
+      // Get payload value
+      az_span payload_span = az_span_init((uint8_t*)message->payload, message->payloadlen);
+      int64_t value_to_double;
+      if ((result = parse_value_to_double(payload_span, &value_to_double)) != AZ_OK)
```

Comment:
I think this is too complicated for a sample. We should not need/use the json parser - we can simply print out the payload.
We can also change to a simpler scenario such as `reboot_device` or `turn_on_led`.

Reply:
Alright I will change it to do a "ping/pong". Method will be named "ping" with no body. Response will always be
```json
{"response":"pong"}
```